### DISCUSSION TITLE
How to disable null-ls for yml files?

### DISCUSSION THREAD

**Discussion by karambaq** [LazyVim/LazyVim#739] [Q&A]
Null-ls truncates 90% of my yml file, when I disable Lsp for buffer - it's ok. How can I disable it for yml files?

**Comment by dpetka2001** [comment]
As far as I know null-ls is used by Lazyvim's formatter. I don't think you can disable null-ls for specific filetypes (at least to my knowledge, i might be wrong), but you can disable autoformat by setting `vim.b.autoformat = false` in an autocmd for yaml files. Maybe that would suit your needs? For an example you could look into #575 which disables autoformat for C++ files. Or you could enforce the null-ls source that formats yaml files to take effect for other files and not yaml, so that it doesn't get used.

**Comment by folke** [comment]
LazyVim does not include formatters for yaml. Not the yamlls lsp and not anything special for null-ls.

So either uninstall whatever null-ls thing you added for formatting yaml files, or disable it in the yamlls settings.

You can easily chck with `:LspInfo` and `:NullLsInfo` what is running in the yaml files

**Reply by karambaq** [reply]
Looks like it's prettierd issue
 
<img width="1490" alt="image" src="https://github.com/LazyVim/LazyVim/assets/14233969/d33e2d14-7771-42f6-bc5b-57594d69f8d6">


<img width="1490" alt="image" src="https://github.com/LazyVim/LazyVim/assets/14233969/a5f8a8b1-0207-40ee-b59c-71bfd0c8a66e">

**Reply by folke** [reply]
Then report an issue there? https://github.com/fsouza/prettierd

**Reply by dpetka2001** [reply]
If it's prettierd just configure its source to not take effect with 
```lua
        nls.builtins.formatting.prettierd.with({
        filetypes = { "html", "css", "json", "jsonc", "markdown" },
        }),
```
Just pass the file extensions that you want prettierd to format.

**Reply by folke** [reply]
yes, but please still file an issue. They won't be able to fix it if they dont know about it

**Reply by karambaq** [reply]
I'm still not sure that it's prettierd. 

as speaking of this:

> LazyVim does not include formatters for yaml. Not the yamlls lsp and not anything special for null-ls.

It is in lazy extras

**Reply by folke** [reply]
Where? There is nothing for yaml in the extras?

**Reply by karambaq** [reply]
@folke I think it's right here:

https://github.com/LazyVim/LazyVim/blob/dfde622847236797d1724988372331f1fbed827b/lua/lazyvim/plugins/extras/formatting/prettier.lua#L5

And it automatically enables yaml formating, I have nothing in my config related to yaml, as soon as I commented this extra formatting for yaml files are gone.

**Reply by dpetka2001** [reply]
That extra is for installing `prettierd` formatter and by default it also formats yaml files. See my comment above to specify which file extensions `prettierd` should format.

**Comment by cerebrl** [comment]
I had the file truncation issue as well, but with Markdown files. I was using the `prettier` extra plugin with a near basic LazyVim configuration.

This [discussion on the `null-ls` project](https://github.com/jose-elias-alvarez/null-ls.nvim/discussions/1609) helped me find [a resolution](https://github.com/jose-elias-alvarez/null-ls.nvim/discussions/1609#discussioncomment-6610361). @caquino seemed to have a good point, saying that `prettierd` alone doesn't seem cause the issue, but a combination of `null-ls` and `prettierd` does.

Anyway, I did as the suggested answer recommends (not using the extra plugin from LazyVim and configuring `prettier` itself instead), and I no longer have this issue. I hope this helps.
